jwst/datamodels/schemas/referencefile.schema.yaml

```diff
@@ -1,32 +1,200 @@
+            enum: [LONG, SHORT, '12', '34', '1', '2', '3', '4', ANY, N/A]
+            fits_keyword: CHANNEL
+          filter:
+            title: Name of the filter element used
+            type: string
+            enum: [ANY, CLEAR, F070LP, F070W, F090W, F1000W, F100LP, F1065C,
+              F110W, F1130W, F1140C, F115W, F1280W, F140M, F140X, F1500W, F150W,
+              F150W2, F1550C, F170LP, F1800W, F182M, F187N, F200W, F2100W, F210M,
+              F212N, F227W, F2300C, F250M, F2550W, F2550WR, F277W, F290LP, F300M,
+              F322W2, F335M, F356W, F360M, F380M, F410M, F430M, F444W,
+              F460M, F480M, F560W, F770W, FLENS, FND, FNDP750L, GR150C,
+              GR150R, OPAQUE, P750L, WL3, WLP4, N/A]
+            fits_keyword: FILTER
+          pupil:
+            title: Name of the pupil element used
+            type: string
+            pattern: "|\
+              ANY|CLEAR|CLEARP|F090W|F115W|F140M|F150W|F158M|F162M|F164N|\
+              F200W|F323N|F405N|F466N|F470N|FLAT|GDHS|GDHS0|GDHS60|GR700XD|\
+              GRISMC|GRISMR|GRISMV2|GRISMV3|MASKBAR|MASKIPR|MASKRND|NRM|PINHOLES|\
+              WLM8|WLP4|WLP8|N/A"
+            fits_keyword: P_PUPIL
+          pupil_imaging_lens:
+            title: Pupil imaging lens in the optical path?
+            type: boolean
+            fits_keyword: PILIN
+          grating:
+            title: Name of the grating element used
+            type: string
+            enum: [G140M, G235M, G395M, G140H, G235H, G395H, PRISM, MIRROR, N/A, ANY]
+            fits_keyword: GRATING
+          band:
+            title: MRS wavelength band
+            type: string
+            enum: [SHORT, MEDIUM, LONG, N/A, ANY]
+            fits_keyword: BAND
+          fixed_slit:
+            title: Name of fixed slit aperture used
+            type: string
+            fits_keyword: FXD_SLIT
+          focus_position:
+            title: Focus position
+            type: number
+            fits_keyword: FOCUSPOS
+          coronagraph:
+            title: coronagraph mask used
+            type: string
+            enum: [4QPM, LYOT, MASK210R, MASKSWB, MASK335R, MASK430R, MASKLWB, NONE]
+            fits_keyword: CORONMSK
+      exposure:
+        title: Exposure parameters
+        type: object
+        properties:
+          p_type:
+            title: Type of data in the exposure
+            type: string
+            pattern: "|\
+              ^FGS_(?=IMAGE|DARK|FOCUS|INTFLAT|SKYFLAT)|\
+              ^MIR_(?=IMAGE|TACQ|LYOT|4QPM|LRS-FIXEDSLIT|LRS-SLITLESS|MRS|DARK|FLAT-IMAGE|FLATIMAGE|FLAT-MRS|FLATMRS|CORONCAL)|\
+              ^NIS_(?=AMI|DARK|FOCUS|IMAGE|LAMP|SOSS|TACQ|TACONFIRM|WFSS)|\
+              ^NRC_(?=IMAGE|GRISM|TACQ|CORON|FOCUS|DARK|FLAT|LED|WFSC|TACONFIRM|TSIMAGE|TSGRISM)|\
+              ^NRS_(?=AUTOFLAT|AUTOWAVE|BOTA|BRIGHTOBJ|CONFIRM|DARK|FIXEDSLIT|FOCUS|IFU|IMAGE|LAMP|MIMF|MSASPEC|TACONFIRM|TACQ|TASLIT)|\
+              ANY|N//A"
+            fits_keyword: P_EXPTYP
```

Comment:
Does this imply that all existing ref files in CRDS will need to have their headers updated to rename the EXP_TYPE keyword to P_EXPTYP?